lib/authc/AssertionAuthenticationResult.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+var nJwt = require('njwt');
+var Account = require('../resource/Account');
+
+function AssertionAuthenticationResult(dataStore, data) {
```

Comment:
The fact that `AuthenticationResult` inherits from `InstanceResource` is weird, and I've wanted to fix that.  I agree that `AuthenticationResult` should be moved to the `lib/authc` folder and not inherit from `InstanceResource`.  Further, `AuthenticationResult` was polluted with methods for client Oauth2 credential authentication, these methods should have been on a new class.  See #357 for more insight into the cleanup that needs to happen.  Of everything I just said, we should work on those issues separately from this PR for SAML.